File: WebSocketsTest/Game/Board.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;
using PetitsChevaux.Plans;
using PetitsChevaux.Plans.MiniMax;
using Random = System.Random;

namespace PetitsChevaux.Game
{
    public class Board
    {
        [UsedImplicitly]
        public static int PawnsPerPlayer = 4;

        [UsedImplicitly]
        public static int PlayerNumber = 4;
        public readonly List<Player> Players = new List<Player>();


        public Board()
        {
            GeneratePlayers();
        }
        public Board(List<Player> players)
        {
            this.Players = players;
        }

        public void GeneratePlayers()
        {
            Players.Clear();
            for (var i = 0; i < PlayerNumber; i++)
            {
                Players.Add(new Player(i)
                {
                    NextMove = ((i == 0) ? (Func<Player, List<Player>, int, Contracts.Action>)NegaMax.NextMove : SimpleMinded.NextMove)
                });
            }
        }
        public Contracts.Action NextTurn(int playerId, int roll)
        {

            int player = Normalize(playerId, Players.Count);
            var result = Players[player].Play(Players, roll);

            return result;
        }

        public static int Normalize(int i, int against = 56, int @base = 0)
        {
            while (true)
            {
                if (i < 0)
                {
                    var t = (against + @base) + i;
                    if (t > 0) return t;
                    i = t;
                    continue;
                }

                if (i >= (against + @base))
                {
                    var t = i - against;
                    if (t < (against + @base)) return t;
                    i = t;
                    continue;
                }

                if (i < (against + @base)) return i;

                break;
            }

            throw new ArgumentOutOfRangeException("i", i, "should be > " + @base);
        }
    }
}

File: PetitsChevauxPy/Game/board.py
from random import randint

def Normalize(i, against = 56, base = 0):
    while(True):
        if(i < 0):
            t = (against + base) + i
            if( t > 0 ) : return t
            i = t;
            continue

        if (i >= (against + base)):
            t = i - against
            if(t < (against + base)) : return t
            i = t
            continue
        if(i < (against + base)) : return i

        break

def RollDice():
    return randint(1,6)


class Board(object):
    PawnsPerPlayer = 4
    PlayerNumber = 4

    

    def __init__(self):
        self.Players = []
        self._lastPlayer = 0
        self.PlayerId = -1

    def NextPlayer(self):
        return Normalize(self._lastPlayer, PlayerNumber)


    def generate_players(self):
        self.Players.clear()
        for x in range(1, PlayerNumber):
            new_p = Player()
            new_p.NextMove = None
            self.Players.append(new_p)






